### PR TITLE
feat: add encryption field

### PR DESCRIPTION
**Pull request by dream2023**
### This is a ...
- [x] New feature
- [ ] Improvement
- [ ] Bug fix
- [ ] Others

### Description 

image

image

image

### Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Add  encryption field|
| 🇨🇳 Chinese | 添加加解密字段|

### Docs

| Language   | Link |
| ---------- | --------- |
| 🇺🇸 English |      |
| 🇨🇳 Chinese |    |

### Checklists
- [ ] All changes have been self-tested and work as expected
- [ ] Test cases are updated/provided or not needed
- [ ] Doc is updated/provided or not needed
- [ ] Component demo is updated/provided or not needed
- [ ] Changelog is provided or not needed
- [ ] Request a code review if it is necessary

### CODE REVIEW

File: packages/plugins/@nocobase/plugin-field-encryption/src/server/utils.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+// 示例用户配置的 Key 和 IV 字符串
+const keyString = '12345678901234567890123456789012'; // 32 字符，适用于 AES-256
+const ivString = '1234567890123456'; // 16 字符，适用于 AES-256-CBC
+
+// 将字符串转换为 Buffer 对象
+const key = Buffer.from(keyString, 'utf8');
+const iv = Buffer.from(ivString, 'utf8');
+
+export function encrypt(text: string) {
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-field-encryption/src/server/utils.ts:21]: encrypt 和 decrypt 要改成 async 的，不然可能会导致事件阻塞

File: packages/plugins/@nocobase/plugin-field-encryption/src/client/index.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * This project is dual-licensed under AGPL-3.0 and NocoBase Commercial License.
+ * For more information, please refer to: https://www.nocobase.com/agreement.
+ */
+
+import { Plugin } from '@nocobase/client';
+import { EncryptionFieldInterface } from './encryption';
+
+export class PluginFieldEncryptionClient extends Plugin {
+  async load() {
+    this.app.dataSourceManager.collectionFieldInterfaceManager.addFieldInterfaces([EncryptionFieldInterface]);
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-field-encryption/src/client/index.tsx:15]: 这个 API，看起来太长了

File: packages/plugins/@nocobase/plugin-field-encryption/src/server/utils.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import crypto from 'crypto';
+const algorithm = 'aes-256-cbc';
+
+// 示例用户配置的 Key 和 IV 字符串
+const keyString = '12345678901234567890123456789012'; // 32 字符，适用于 AES-256
+const ivString = '1234567890123456'; // 16 字符，适用于 AES-256-CBC
+
+// 将字符串转换为 Buffer 对象
+const key = Buffer.from(keyString, 'utf8');
+const iv = Buffer.from(ivString, 'utf8');
```
Comment (chenos) [inline packages/plugins/@nocobase/plugin-field-encryption/src/server/utils.ts:19]: 这里的做法看起来也不太好，AES 可以和 RSA 结合起来

File: packages/plugins/@nocobase/plugin-field-encryption/src/server/utils.ts (hunk 1 above)
Comment (dream2023) [reply packages/plugins/@nocobase/plugin-field-encryption/src/server/utils.ts:21]: image

API 中好像没有在读字段值前能做处理的 Hook @chenos

**Pull request closed without merging**